Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -826,6 +861,66 @@ Timestamp Serializer::DecodeTimestamp(
   return Timestamp{timestamp_proto.seconds, timestamp_proto.nanos};
 }
 
+/* static */
+google_type_LatLng Serializer::EncodeGeoPoint(const GeoPoint& geo_point_value) {
+  google_type_LatLng result{};
+  result.latitude = geo_point_value.latitude();
+  result.longitude = geo_point_value.longitude();
+  return result;
+}
+
+/* static */
+GeoPoint Serializer::DecodeGeoPoint(nanopb::Reader* reader,
+                                    const google_type_LatLng& latlng_proto) {
+  // The GeoPoint ctor will assert if we provide values outside the valid range.
+  // However, since we're decoding, a single corrupt byte could cause this to
+  // occur, so we'll verify the ranges before passing them in since we'd rather
+  // not abort in these situations.
+  double latitude = latlng_proto.latitude;
+  double longitude = latlng_proto.longitude;
+  if (std::isnan(latitude) || latitude < -90 || 90 < latitude) {
```

Comment:
Question: I presume using an epsilon value for comparison here would be overkill?